librz/search/search_internal.h

```diff
@@ -0,0 +1,43 @@
+// SPDX-FileCopyrightText: 2024 RizinOrg <info@rizin.re>
+// SPDX-FileCopyrightText: 2024 deroad <wargio@libero.it>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#ifndef RZ_SEARCH_INTERNAL_H
+#define RZ_SEARCH_INTERNAL_H
+
+#include <rz_search.h>
+#include <rz_list.h>
+#include <rz_th.h>
+
+#define RZ_SEARCH_AES_LENGTH         40
+#define RZ_SEARCH_PRIVATE_KEY_LENGTH 11
+#define RZ_SEARCH_MIN_BUFFER_SIZE    512u
+#define RZ_SEARCH_MAX_HEX_PATTERN    UT16_MAX
+
+typedef void (*RzSearchFiniCallback)(void *user);
+typedef bool (*RzSearchOverCallback)(RzPVector /*<void *>*/ *collection, ut64 address, const ut8 *buffer, size_t size, RzThreadQueue *hits);
+
+struct rz_search_collection_t {
+	RzPVector /*<void *>*/ *collection; ///< Collection of elements to search in a buffer
+	RzSearchOverCallback search_over; ///< Collection search over the collection callback
```

Comment:
Docs are a little confusing. Is the `search_over` callback called for every elementin the collection above?

Reply:
No, the function takes the vector and uses it